v23.1/full-text-search.md

```diff
@@ -0,0 +1,470 @@
+You can rank the results of a full-text search.
+
+The `ts_rank()` [built-in function](functions-and-operators.html#full-text-search-functions) outputs a search rank based on the frequency of matching lexemes. In the following example, two lexemes match:
+
+~~~ sql
+SELECT ts_rank(to_tsvector('How do trees get on the internet?'), plainto_tsquery('how to get internet'));
+~~~
+
+{% include_cached copy-clipboard.html %}
+~~~
+   ts_rank
+--------------
+  0.09735848
+~~~
+
+In this example, three lexemes match, resulting in a higher rank:
+
+~~~ sql
+SELECT ts_rank(to_tsvector('How do trees get on the internet?'), plainto_tsquery('wow, do trees get internet?'));
+~~~
+
+{% include_cached copy-clipboard.html %}
+~~~
+   ts_rank
+--------------
+  0.26426345
+~~~
+
+{{site.data.alerts.callout_info}}
+Because a rank must be calculated for each matching document, ranking a full-text search can incur a performance overhead if there are many matching documents.
+{{site.data.alerts.end}}
+
+## Comparisons
+
+Full-text searches support the following comparison operator:
+
+- **matching**: [`@@`](functions-and-operators.html#operators). This operator is set between a `TSQUERY` and `TSVECTOR`, and returns `true` if the lexemes match. The `TSQUERY` and `TSVECTOR` can be specified in any order.
+
+For usage examples, see [Match queries to documents](#match-queries-to-documents).
+
+## Full-text indexes
+
+{{site.data.alerts.callout_info}}
+You can perform full-text searches without a full-text index. However, an index can improve search performance. 
+{{site.data.alerts.end}}
+
+To create a full-text index, use the [`CREATE INDEX`](create-index.html) syntax that defines an [inverted index](inverted-indexes.html), specifying a `TSVECTOR` column.
+
+- Using the PostgreSQL-compatible syntax:
+
+    ~~~ sql
+    CREATE INDEX {optional name} ON {table} USING GIN ({column});
+    ~~~
+
+    {{site.data.alerts.callout_info}}
+    GIN and GiST indexes are implemented identically on CockroachDB. `GIN` and `GIST` are therefore synonymous when defining a full-text index.
+    {{site.data.alerts.end}}
+
+- Using `CREATE INVERTED INDEX`:
+
+    ~~~ sql
+    CREATE INVERTED INDEX {optional name} ON {table} ({column});
+    ~~~
+
+For more ways to define full-text indexes, see [Create a full-text index with an expression](#create-a-full-text-index-with-an-expression) and [Create a full-text index with a stored computed column](#create-a-full-text-index-with-a-stored-computed-column).
+
+## Text search configuration
+
+A *text search configuration* determines how inputs are parsed into `TSVECTOR` and `TSQUERY` values. This includes a dictionary that is used to identify derivatives of words, as well as stop words to exclude, when normalizing [documents](#process-a-document) and [queries](#form-a-query).
+
+The currently supported dictionaries are English, Danish, Dutch, Finnish, French, German, Hungarian, Italian, Norwegian, Portuguese, Russian, Spanish, Swedish, and Turkish.
```

Comment:
There's also the "simple" dictionary which doesn't include any stemming or stopwording.